src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentState.cs

```diff
@@ -117,22 +117,11 @@ protected static ValueSource<TextAndVersion> CreateRecoverableText(TextLoader lo
                     asynchronousComputeFunction: cancellationToken => loader.LoadTextAsync(services.Workspace, documentId, cancellationToken),
                     synchronousComputeFunction: cancellationToken => loader.LoadTextSynchronously(services.Workspace, documentId, cancellationToken),
                     cacheResult: false),
-                services.GetRequiredService<ITemporaryStorageServiceInternal>());
+                services);
         }
 
         public ITemporaryTextStorageInternal? Storage
-        {
-            get
-            {
-                var recoverableText = this.TextAndVersionSource as RecoverableTextAndVersion;
-                if (recoverableText == null)
-                {
-                    return null;
-                }
-
-                return recoverableText.Storage;
```

Comment:
note to self.  Have an analyzer that finds thsi pattern and recommends the new form.